### PR TITLE
Fixed compiler warnings, function arguments and const parameter

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot be nice and test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: include/freerdp/version.h.in [hunk 1]
```diff
@@ -27,6 +27,6 @@
 #define FREERDP_API_VERSION "${FREERDP_API_VERSION}"
 #define FREERDP_VERSION "${FREERDP_VERSION}"
 #define FREERDP_VERSION_FULL "${FREERDP_VERSION_FULL}"
-#define GIT_REVISION "${GIT_REVISION}"
+#define FREERDP_GIT_REVISION "${GIT_REVISION}"
```
Comment (hardening) [inline include/freerdp/version.h.in:30]: Looks like an API breakage...

**Comment by hardening** [comment]
I agree the `GIT_REVISION` define was unfortunate, but it was public, if we remove it let's hope no one was using it...

**Comment by akallabeth** [comment]
@hardening we´ll see and this is master, so breakage expected. Will of course, not be backported :)

**Pull request merged**